src/trans.rs

```diff
@@ -314,12 +314,12 @@ impl<'v, 'tcx: 'v> BinaryenFnCtxt<'v, 'tcx> {
                     let a = self.trans_operand(a);
                     let b = self.trans_operand(b);
                     let op = match *op {
-                        BinOp::Add => BinaryenAdd(),
-                        BinOp::Sub => BinaryenSub(),
-                        BinOp::Mul => BinaryenMul(),
-                        BinOp::Div => BinaryenDivS(),
-                        BinOp::Eq => BinaryenEq(),
-                        BinOp::Ne => BinaryenNe(),
+                        BinOp::Add => BinaryenAddInt32(),
```

Comment:
Oh, I didn't notice the panic on non-i32. Sounds fine for now then, I think. A `TODO` sounds good though, would make it clear what next steps are.